### ISSUE TITLE
[mdc-floating-label] Prefilled mdc-texfield and mdc-select animates the label when browser refresh

### ISSUE THREAD

**Issue by satlom** [material-components/material-components-web#7002]
## Bug report
When a page reload and have made your select prefilled with a value, the label will animate.

### Steps to reproduce
Go to page
https://material-components.github.io/material-components-web-components/demos/select/

refresh the page and noticed the prefilled selected label animate.
I know this is not this repo, but I see same behavior in my code.

### Actual behavior
The label moves when select is prefilled.

### Expected behavior
The label should not move when prefilled. In cases of server pre-rendering it looks weird.

### Your Environment:

| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          | 10
| Browser          | Chrome -version 89.0.4389.128
| Operating System | Windows 10

**Comment by EstebanG23** [comment]
Hi there, thank you for reporting this bug.

The demo you provided uses MWC components from https://github.com/material-components/material-components-web-components. If your use case is from these components I'd suggest filing this issue on that repository for better assistance.

If you are using MDC select and are in charge of rendering, then you can add `mdc-floating-label--float-above` to the floating label and `mdc-notched-outline--notched` to the notched outline whenever there is preselected text. 

Please let me know if this helps.

**Comment by satlom** [comment]
Hi!
I changed the title on this bugreport!
I have attached a sample project to show what's going on.
[mdc-textbox.zip](https://github.com/material-components/material-components-web/files/6364159/mdc-textbox.zip)
Unpack the ZIP file and run "nmp i", "npm run build" and "npm run start". Open a browser on port 3000.
One thing I noticed, was that I got the same result even if I skipped the init of the component in javascript.
This mean that it probably related to the CSS (SASS) and not the javascript.

**Comment by patrickrodee** [comment]
We have [docs](https://github.com/material-components/material-components-web/tree/master/packages/mdc-textfield#pre-filled) for this case. Let me know if those don't help.

**Comment by satlom** [comment]
Hi patrickrodee!
I have followed the documentation for the attached example. Could you please point out what is wrong or missing?

**Comment by patrickrodee** [comment]
Without seeing your demo, it's difficult for me to say. Can you reproduce the issue in one of our starter kits?

- Glitch: https://glitch.com/edit/#!/remix/new-web
- Codepen: https://codepen.io/abhiomkar/pen/gQWarJ

**Comment by satlom** [comment]
The project is based on a starterkit.
The problem with the glitch and codepen is that it is hard to do a full browser reload and see the effect.
This is what happens when you do a server pre-render and later on load the javascript. In my case I use Blazor Client, but the pure html+javascript example shows what happens.

**Comment by satlom** [comment]
I made further investigation about this issue.
What I found out was when a load my page with my script tag bundle.js and with the async tag to it, it didn't work. The label flashes.
With different browsers I saw different results:
Latest Chrome = label flashes
Latest Firefox = the width of the textbox flashes

When I remove the async tag everything works as expected.
I believe this has to something to do with FOUC (Flash Of Unstyled Content) and is connected to the environment you are in.
I close this issue now, and I don't think it is related to material-components-web.
